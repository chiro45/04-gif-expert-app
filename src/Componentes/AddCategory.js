import React, {useState} from 'react'
import PropTypes from 'prop-types';
export default function AddCategory({setCategories}) {
    
    const [inputValue, setinputValue] = useState("");
    const handleInputChange = (e)=>{
        setinputValue(e.target.value)
    }
    const handleSubmit = (e)=>{
        e.preventDefault(e)
        console.log('handle submit', inputValue)

        if(inputValue.trim().length > 2){
            
         setCategories( categories => [inputValue, ...categories,])
         setinputValue('')

        }
        //console.log('submit hecho')
    }
    return (      
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
        <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
        />
        </form>   
    )
}
AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}