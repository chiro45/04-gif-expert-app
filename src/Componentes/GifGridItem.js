import React from 'react'
import PropTypes from 'prop-types';
export const GifGridItem = ({ title, img}) => {
    
    return (
        <div className='card animate__animated animate__backInRight'>
            <img src={img} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,

}