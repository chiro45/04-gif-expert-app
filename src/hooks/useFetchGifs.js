
import {useState, useEffect} from 'react';
import getGifs from '../helpers/getGifs'

export const useFethGifs = ( category )=>{
    const [state, setState] = useState({
        data: [],
        loading: true,

    });
     useEffect(()=>{
         //ejecutamos la funcion
            getGifs(category)
            //luego las imagenes van al seState
            .then(imgs=>{setState({
                data: imgs,
                loading: false,
            });
        })
       //arreglo de dependencias
     },[category])

    return state; // {data, loading}
}