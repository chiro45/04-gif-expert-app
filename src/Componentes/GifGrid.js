import React from 'react'
import { useFethGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types'

export const GifGrid = ({category}) => {
    const {data:images, loading} = useFethGifs(category);
    return (
        <>
        <h3 animate__animated animate__backInRight>{category}</h3>
        {loading && <p className='loading animate__animated animate__backInLeft '>Loading...</p>}
        <div className='card-grid animate__animated animate__backInRight'>   
                {
                    images.map( img =>(
                       <GifGridItem
                       key={img.id}
                       {...img}
                       />
                    ))
                }
        </div> 
        </>
    )
}
GifGrid.propTypes ={
    category: PropTypes.string.isRequired
}