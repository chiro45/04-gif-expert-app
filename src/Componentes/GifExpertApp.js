import React, {useState} from 'react';
import AddCategory from './AddCategory';
import PropTypes from 'prop-types';
import { GifGrid } from './GifGrid';


export const GifExpertApp = () =>{
    
    const [categories, setCategories] = useState( ['Avengers']);
    // const handleAdd = ()=>{
    //    setcategories([...categories, "juan CArlo"])
    // }
    
    // const categories = ['one punch', 'samurai X', 'dragon Ball'];
    return ( 
        <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories ={setCategories}/>
        <hr/>
        <ol>

        {

        categories.map(category => (
                       < GifGrid 
                       key={category}
                       category={category}/>
                    )
        
        )}

        </ol>
        
        </>
     );

    
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}


