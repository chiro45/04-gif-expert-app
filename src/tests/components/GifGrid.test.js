import React from 'react';
import { shallow } from "enzyme";
import { GifGrid } from "../../Componentes/GifGrid";
import { useFethGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')


describe('Pruebas en el <GifGrid />', () => {

    const category = 'One Punch';

    test('debe de mostrarse correctamente', () => {

        useFethGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category } /> );
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            img: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: '123',
            img: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }];

        useFethGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        // expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);

        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    })
      
})
