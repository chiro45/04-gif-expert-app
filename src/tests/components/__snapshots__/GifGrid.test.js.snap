// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    animate__animated={true}
    animate__backInRight={true}
  >
    One Punch
  </h3>
  <p
    className="loading animate__animated animate__backInLeft "
  >
    Loading...
  </p>
  <div
    className="card-grid animate__animated animate__backInRight"
  />
</Fragment>
`;
